File: ray_tracing/camera.py
from ray_tracing.ray import Ray
from ray_tracing.vector import vector


class Camera(object):
    def __init__(self, origin):
        self.origin = origin

    def ray_to_position(self, x, y):
        position = vector(2 * x - 1, 2 * y - 1, 0)

        return Ray(self.origin, position - self.origin)

File: ray_tracing_rust/src/camera.rs
use crate::ray::Ray;
use crate::vector::Vector3;

pub struct Camera {
    origin: Vector3,
}

impl Camera {
    pub fn new(origin: Vector3) -> Self {
        Self { origin }
    }
}

impl Camera {
    pub fn position_ray(&self, u: f32, v: f32) -> Ray {
        let pixel_position = Vector3 {
            x: u * 2_f32 - 1_f32,
            y: v * 2_f32 - 1_f32,
            z: 0_f32,
        };

        Ray::new(self.origin, pixel_position - self.origin)
    }
}
